packages/package-manager/src/NpmPackageManager.ts

```diff
@@ -0,0 +1,179 @@
+  `${ansi}npm${ansi} ${ansi}WARN${ansi}.+You must install peer dependencies yourself\\.\n`,
+  'g'
+);
+
+/** Exposed for testing */
+export class NpmStderrTransform extends Transform {
+  _transform(
+    chunk: Buffer,
+    encoding: string,
+    callback: (error?: Error | null, data?: any) => void
+  ) {
+    this.push(chunk.toString().replace(npmPeerDependencyWarningPattern, ''));
+    callback();
+  }
+}
+
+export class NpmPackageManager implements PackageManager {
+  options: SpawnOptions;
+
+  private log: Logger;
+
+  constructor({ cwd, log, silent }: { cwd: string; log?: Logger; silent?: boolean }) {
+    this.log = log || console.log;
+    this.options = {
+      env: {
+        ...process.env,
+        ...DISABLE_ADS_ENV,
+      },
+      cwd,
+      ...(silent
+        ? { ignoreStdio: true }
+        : {
+            stdio: ['inherit', 'inherit', 'pipe'],
+          }),
+    };
+  }
+
+  get name() {
+    return 'npm';
+  }
+
+  async installAsync(parameters: string[] = []) {
+    await this._runAsync(['install', ...parameters]);
+  }
+
+  async addGlobalAsync(...names: string[]) {
+    if (!names.length) return this.installAsync();
+    await this._runAsync(['install', '--global', ...names]);
+  }
+
+  async addWithParametersAsync(names: string[], parameters: string[] = []) {
+    if (!names.length) return this.installAsync(parameters);
+
+    const { versioned, unversioned } = this._parseSpecs(names);
+    if (versioned.length) {
+      await this._patchAsync(versioned, 'dependencies');
+      await this.installAsync(parameters);
+    }
+    if (unversioned.length) {
+      await this._runAsync([
+        'install',
+        '--save',
+        ...unversioned.map(spec => spec.raw),
+        ...parameters,
+      ]);
+    }
+  }
+
+  async addAsync(...names: string[]) {
+    await this.addWithParametersAsync(names, []);
+  }
+
+  async addDevAsync(...names: string[]) {
+    if (!names.length) return this.installAsync();
+
+    const { versioned, unversioned } = this._parseSpecs(names);
+    if (versioned.length) {
+      await this._patchAsync(versioned, 'devDependencies');
+      await this._runAsync(['install']);
+    }
+    if (unversioned.length) {
+      await this._runAsync(['install', '--save-dev', ...unversioned.map(spec => spec.raw)]);
+    }
+  }
+
+  async removeAsync(...names: string[]) {
+    await this._runAsync(['uninstall', ...names]);
+  }
+
+  async versionAsync() {
+    const { stdout } = await spawnAsync('npm', ['--version'], { stdio: 'pipe' });
+    return stdout.trim();
+  }
+
+  async getConfigAsync(key: string) {
+    const { stdout } = await spawnAsync('npm', ['config', 'get', key], { stdio: 'pipe' });
+    return stdout.trim();
+  }
+
+  async removeLockfileAsync() {
+    assert(this.options.cwd, 'cwd required for NpmPackageManager.removeLockfileAsync');
+    const lockfilePath = path.join(this.options.cwd, 'package-lock.json');
+    if (fs.existsSync(lockfilePath)) {
+      rimraf.sync(lockfilePath);
```

Comment:
existsSync does not have a good async equivalent in fs/promise, but why `rimraf` call is sync inside async function?

Reply:
Yeah, I think @dsokal mentioned getting rid of rimraf. We can definitely swap `rimraf` for just `fs` and use async methods. Can follow up in a later PR after https://github.com/expo/expo/pull/18576